Revised user model and added info to JWT:

jovanlekic opened the pull request:
- changed UserDetailsImpl ( added missing fields from User model)
 - changed SignupRequest ( added missing fields from User and changed user-role relationship)
 - changed User ( added missing fields and constraints)
 - added generating JWT from UserDetailsImpl
 - changed migration scripts to correspond with database design

DusanBrankovic committed:
Created ExchangeRate class. Created migration file for exchange_rates table.

DusanBrankovic committed:
Created ExchangeRate class. Created migration file for exchange_rates table.

DusanBrankovic committed:
Created ExchangeRate class. Created migration file for exchange_rates table.

DusanBrankovic committed:
Created ExchangeRate class. Created migration file for exchange_rates table.

DusanBrankovic committed:
Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate.

DusanBrankovic committed:
Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate.

DusanBrankovic committed:
Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate.

DusanBrankovic committed:
Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate.

DusanBrankovic committed:
Removed @Unsigned annotation from amountInRsd property in ExchangeRate class. Changed ExchangeRateResponse to match with ExchangeRate.

DusanBrankovic committed:
Created request DTO for ExchangeRate.

DusanBrankovic committed:
Created request DTO for ExchangeRate.

DusanBrankovic committed:
Created request DTO for ExchangeRate.

DusanBrankovic committed:
Created request DTO for ExchangeRate.

DusanBrankovic committed:
Created Mapper class for ExchangeRate and its DTOs

DusanBrankovic committed:
Created ExchangeRateController with shell endpoints.

DusanBrankovic committed:
Renamed ExchangeRateRequest to ExchangeRatePostRequest. Created ExchangeRatePutRequest.

DusanBrankovic committed:
Renamed ExchangeRateRequest to ExchangeRatePostRequest. Created ExchangeRatePutRequest.

DusanBrankovic committed:
Renamed ExchangeRateRequest to ExchangeRatePostRequest. Created ExchangeRatePutRequest.

DusanBrankovic committed:
Renamed ExchangeRateRequest to ExchangeRatePostRequest. Created ExchangeRatePutRequest.

DusanBrankovic committed:
Renamed ExchangeRateRequest to ExchangeRatePostRequest. Added attributes in ExchangeRatePutRequest.

DusanBrankovic committed:
Injected ExchangeRateService in ExchangeRateController.

DusanBrankovic committed:
Changed mapExchangeRateRequestToExchangeRate -> mapExchangeRatePostRequestToExchangeRate. Added mapping from ExchangeRatePutRequest to ExchangeRate.

DusanBrankovic committed:
Revert "Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate."

This reverts commit 0dce1a68

DusanBrankovic committed:
Revert "Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate."

This reverts commit 0dce1a68

DusanBrankovic committed:
Revert "Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate."

This reverts commit 0dce1a68

DusanBrankovic committed:
Revert "Changed property names in ExchangeRate class and migration file to more accurately represent what info they hold. Changed hibernate property from update to validate."

This reverts commit 0dce1a68

DusanBrankovic committed:
Removed migration script for ExchangeRate model.

DusanBrankovic committed:
Removed migration script for ExchangeRate model.

DusanBrankovic committed:
Removed migration script for ExchangeRate model.

DusanBrankovic committed:
Removed migration script for ExchangeRate model.

jovanlekic committed:
Revised user model and added info to JWT:
 - changed UserDetailsImpl ( added missing fields from User model)
 - changed SignupRequest ( added missing fields from User and changed user-role relationship)
 - changed User ( added missing fields and constraints)
 - added generating JWT from UserDetailsImpl
 - changed migration scripts to correspond with database design

jovanlekic committed:
Revised user model and added info to JWT:
 - changed UserDetailsImpl ( added missing fields from User model)
 - changed SignupRequest ( added missing fields from User and changed user-role relationship)
 - changed User ( added missing fields and constraints)
 - added generating JWT from UserDetailsImpl
 - changed migration scripts to correspond with database design

jovanlekic committed:
Revised user model and added info to JWT:
 - changed UserDetailsImpl ( added missing fields from User model)
 - changed SignupRequest ( added missing fields from User and changed user-role relationship)
 - changed User ( added missing fields and constraints)
 - added generating JWT from UserDetailsImpl
 - changed migration scripts to correspond with database design

jovanlekic committed:
Applied spotless changes and upgraded google java format version

jovanlekic committed:
Applied spotless changes and upgraded google java format version

jovanlekic committed:
Applied spotless changes and upgraded google java format version

megicivovic commented on travel/src/main/java/com/tx/travel/controller/AuthController.java:
```diff
@@ -95,13 +102,18 @@ public ResponseEntity<?> registerUser(@Valid @RequestBody final SignupRequest si
+    final String strRole = signUpRequest.getRole();
 
-    final Set<Role> roles = authService.addRoles(strRoles);
+    final Role role = authService.addRole(strRole); // mapiram String -> Role
```
remove comment pls

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -1,59 +1,77 @@
+  @Size(min = 8, max = 9)
+  private String id_no;
+
+  @NotNull private Long cost_center_id;
```
use camel case for variable names everywhere

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
+    this.surname = surname;
+  }
+
+  public String getPassport_no() {
```
camel case im method names please!

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
+    return passport_no;
+  }
+
+  public void setPassport_no(String passport_no) {
```
camel case please!

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
+    this.passport_no = passport_no;
+  }
+
+  public String getId_no() {
```
camel case im method names please!

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
+    return id_no;
+  }
+
+  public void setId_no(String id_no) {
```
camel case im method names please!

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
+    this.id_no = id_no;
+  }
+
+  public Long getCost_center_id() {
```
camel case im method names please!

megicivovic commented on travel/src/main/java/com/tx/travel/model/User.java:
```diff
@@ -88,11 +106,51 @@ public void setPassword(String password) {
 
-  public void setRoles(Set<Role> roles) {
-    this.roles = roles;
+  public void setCost_center_id(Long cost_center_id) {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
 
+  @NotBlank
+  @Size(min = 8, max = 10)
+  private String passport_no;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+
+  @NotBlank
+  @Size(min = 8, max = 9)
+  private String id_no;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+  @Size(max = 20)
+  private String surname;
+
+  @NotNull private Long cost_center_id;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+
+  @NotNull private Long cost_center_id;
+
+  public String getPassport_no() {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+    return passport_no;
+  }
+
+  public void setPassport_no(String passport_no) {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+    this.passport_no = passport_no;
+  }
+
+  public String getId_no() {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+    return id_no;
+  }
+
+  public void setId_no(String id_no) {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+    this.surname = surname;
+  }
+
+  public Long getCost_center_id() {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +15,70 @@ public class SignupRequest {
+    return cost_center_id;
+  }
+
+  public void setCost_center_id(Long cost_center_id) {
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java:
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
   private String username;
   private String email;
+
+  private String passport_no;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java:
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
   private String email;
+
+  private String passport_no;
+  private String id_no;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java:
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
+  private String id_no;
+  private String name;
+  private String surname;
+  private Long cost_center_id;
```
camel case in method and variable names please!

megicivovic commented on travel/src/main/java/com/tx/travel/payload/response/UserInfoResponse.java:
```diff
@@ -6,15 +6,76 @@ public class UserInfoResponse {
     this.roles = roles;
   }
 
+  public String getPassport_no() {
```
camel case in method and variable names please! - apply to all methods and variables in all classes

megicivovic commented on travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java:
```diff
@@ -40,7 +42,15 @@ public String getJwtFromCookies(HttpServletRequest request) {
+    String jwt = generateTokenFromUser(userPrincipal);
+    return ResponseCookie.from(jwtCookie, jwt)
+        .path("/api")
+        .maxAge(24 * 60 * 60L)
```
can we make this configurable?

jovanlekic replied:
Can you be more precise? Do you mean adding more properties to response cookie, as '.secure()' for example or changing path to be less general "./api/..."

megicivovic replied:
the comment is referring to line .maxAge,so I mean this number should be part of application, properties and injected here as a value. But now that you mentioned it, "/api" should be a constant at least, if not also part of configuration. Always avoid using hard coded values!

megicivovic commented on travel/src/main/java/com/tx/travel/security/jwt/JwtUtils.java:
```diff
@@ -75,9 +86,18 @@ public boolean validateJwtToken(String authToken) {
 
-  public String generateTokenFromUsername(String username) {
+  public String generateTokenFromUser(UserDetailsImpl userPrincipal) {
+    Map<String, Object> claims = new HashMap<>();
```
can they be String?

jovanlekic replied:
.setClaims() from Jwts package wouldn't be able to resolve Map<String,String> , accepts only Map<String, Object>

megicivovic commented on travel/src/main/java/com/tx/travel/security/services/UserDetailsImpl.java:
```diff
@@ -19,6 +19,12 @@ public class UserDetailsImpl implements UserDetails {
 
   private String email;
 
+  private String passport_no;
```
camel case in method and variable names please!

jovanlekic committed:
Applied changes suggester by comments on CR : camelCase on variables and methods name, changed migration scripts version

jovanlekic committed:
Applied changes suggester by comments on CR : camelCase on variables and methods name, changed migration scripts version

jovanlekic committed:
Applied changes suggester by comments on CR : camelCase on variables and methods name, changed migration scripts version

DusanBrankovic committed:
Continued implementation of exchange rate endpoint

* Changes in ExchangeRate:
  - changed `amountInRsd` type from Float to BigDecimal for better control over data
  - renamed `currency` to `code`
  - removed @NotNull annotation from `code` field since @NotBlank covers not null constraint
  - added @Builder, @AllArgsConstructor and @NoArgsConstructor lombok annotations

* Changes in ExchangeRateController:
  - added apiPrefix property to application.properties, and referenced it in @RequestMapping for this endpoint
  - injected ExchangeRateMapper through constructor (it was previously in ExchangeRateServiceLocalImpl)
  - implemented methods for CRUD operations (getAll, getById, create, update, delete)
  - added @Valid annotations to method arguments to check if received data is valid

* Changes in ExchangeRateMapper:
  - added @Component annotation so the class is auto-detected
  - added `id` in mapExchangeRatePutRequestToExchangeRate method signature

* Changes in ExchangeRatePostRequest:
  - added @AllArgsConstructor and @NoArgsConstructor lombok annotations
  - changed `amountInRsd` type from Float to BigDecimal
  - added validation for fields

* Changes in ExchangeRatePutRequest:
  - added @AllArgsConstructor and @NoArgsConstructor lombok annotations
  - changed `amountInRsd` type from Float to BigDecimal
  - added validation for fields

* Changes in ExchangeRateResponse:
  - changed `amountInRsd` type from Float to BigDecimal

+ Created ExchangeRateAlreadyDefined exception
+ Created ExchangeRateNotFound exception
+ Created ExchangeRateRepository
+ Created ExchangeRateService interface
+ Created ExchangeRateServiceLocalImpl:
  - implemented methods for CRUD operations (getAll, getById, create, update, delete)

DusanBrankovic committed:
Continued implementation of exchange rate endpoint

* Changes in ExchangeRate:
  - changed `amountInRsd` type from Float to BigDecimal for better control over data
  - renamed `currency` to `code`
  - removed @NotNull annotation from `code` field since @NotBlank covers not null constraint
  - added @Builder, @AllArgsConstructor and @NoArgsConstructor lombok annotations

* Changes in ExchangeRateController:
  - added apiPrefix property to application.properties, and referenced it in @RequestMapping for this endpoint
  - injected ExchangeRateMapper through constructor (it was previously in ExchangeRateServiceLocalImpl)
  - implemented methods for CRUD operations (getAll, getById, create, update, delete)
  - added @Valid annotations to method arguments to check if received data is valid

* Changes in ExchangeRateMapper:
  - added @Component annotation so the class is auto-detected
  - added `id` in mapExchangeRatePutRequestToExchangeRate method signature

* Changes in ExchangeRatePostRequest:
  - added @AllArgsConstructor and @NoArgsConstructor lombok annotations
  - changed `amountInRsd` type from Float to BigDecimal
  - added validation for fields

* Changes in ExchangeRatePutRequest:
  - added @AllArgsConstructor and @NoArgsConstructor lombok annotations
  - changed `amountInRsd` type from Float to BigDecimal
  - added validation for fields

* Changes in ExchangeRateResponse:
  - changed `amountInRsd` type from Float to BigDecimal

+ Created ExchangeRateAlreadyDefined exception
+ Created ExchangeRateNotFound exception
+ Created ExchangeRateRepository
+ Created ExchangeRateService interface
+ Created ExchangeRateServiceLocalImpl:
  - implemented methods for CRUD operations (getAll, getById, create, update, delete)

jovanlekic committed:
Writing more secure variant of signup:
- currently disabled option to pass role in DTO
because someone can inject different role,
so for every signup request user will get
ROLE_EMPLOYEE role and leave past addRole
to be accesible only for ADMIN CMS

Writing unit tests for AuthService

jovanlekic committed:
Writing more secure variant of signup:
- currently disabled option to pass role in DTO
because someone can inject different role,
so for every signup request user will get
ROLE_EMPLOYEE role and leave past addRole
to be accesible only for ADMIN CMS

Writing unit tests for AuthService

jovanlekic committed:
Put constants to config file to avoid hard coding

jovanlekic committed:
Put constants to config file to avoid hard coding

DusanBrankovic committed:
Removed validateByCode method from interface.

jovanlekic committed:
Merge branch 'master' into feature/jwt_and_user_model

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+
+        final String role = "mod";
+
+        final Role rola = new Role(ERole.ROLE_OFFICE_MANAGER);
```
use english names

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void findByUsernameOrEmail_thrownEmailAlreadyExistedException_success(){
```
```suggestion
    public void findByUsernameOrEmail_thrownEmailAlreadyExistedException(){
```

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    @InjectMocks
+    AuthService sut;
+
+    @DisplayName("given auth service" +
```
rephrase please, given is condition, then is expected result

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void addRoleAdmin_retAdmin_success(){
```
```suggestion
    public void addRoleAdmin_success(){
```

megicivovic replied:
add display name

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void addRole_retOM_success(){
```
```suggestion
    public void addRoleOM_success(){
```

megicivovic replied:
add display name

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void addRole_retEmpl_success(){
```
add display name

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    @Test
+    public void addRole_retEmpl_success(){
+
+        final String role = "afa"; //nesto sto nije admin/mod
```
```suggestion
        final String role = "test_role";
```

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+
+        final String role = "afa"; //nesto sto nije admin/mod
+
+        final Role rola = new Role(ERole.ROLE_EMPLOYEE);
```
english names

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    public void addRole_retEmplNull_success(){
```
```suggestion
    public void addRole_null{
```

megicivovic replied:
add display name for this test

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+    @Test
+    public void addRole_retEmplNull_success(){
+
+        final String role = null;//nesto sto nije admin/mod
```
```suggestion
        final String role = null;
```

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -0,0 +1,132 @@
+
+        final String role = null;//nesto sto nije admin/mod
+
+        final Role rola = new Role(ERole.ROLE_EMPLOYEE);
```
english names

DusanBrankovic committed:
Added methods (validateByCode, validateCodeContent) to validate currency code in ExchangeRateServiceLocalImpl.

Removed validateByCode() method from ExchangeRateService interface.

Renamed ExchangeRateNotFound to ExchangeRateNotFoundException.
Renamed ExchangeRateAlreadyDefined to ExchangeRateAlreadyDefinedException.

Created new ExchangeRateCodeNotValidException.
Created new ExchangeRateAmountInRsdNotValidException.
Created ExchangeRateServiceLocalImplTest.

DusanBrankovic committed:
Removed unused Exceptions.

DusanBrankovic committed:
Changes from PR

* Changes in ExchangeRateServiceLocalImplTest:
  - Removed star imports from ExchangeRateServiceLocalImplTest.
  - Corrected "given" for getExchangeRateById_success and getExchangeRateById_notFound methods.

* Changes in ExchangeRateServiceLocalImpl:
  - Changed deleteExchangeRate implementation to avoid handling DB exception.
  - Changed getExchangeById method to throw ExchangeRateNotFoundException instead of NoSuchElementException.

* Changes in ExchangeRateServiceLocalImplTest:
  - Changed tests for deleteExchangeRate_success and deleteExchangeRate_exchangeRateNotFound to align with new implementation of deleteExchangeRate method in ExchangeRateServiceLocalImpl.

* Changed caught Exception from NoSuchElementException to ExchangeRateNotFoundException in getExchangeRateById.

* Formatting in ExchangeRate class.

petkocfc commented on travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java:
```diff
@@ -15,12 +16,72 @@ public class SignupRequest {
 
+  @NotBlank
+  @Size(min = 8, max = 10)
+  @Column(name="passport_no")
```
Why do we have @Column in Request?

petkocfc commented on travel/src/main/java/com/tx/travel/service/AuthService.java:
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
-      final Role userRole =
-          roleRepository
+  public Role addRoleAdmin(final String strRole) {
+    if (strRole == null) {
```
this method should be refactored

petkocfc commented on travel/src/main/java/com/tx/travel/service/AuthService.java:
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
-    return roles;
+  }
+
+  public Role addRole(final ERole role) {
```
here we have a method addRole and we are fetching the role, shouldn't it be `getRole` or?

petkocfc commented on travel/src/main/resources/db/migration/V4__drop_user_roles.sql:
```diff
@@ -0,0 +1 @@
+DROP TABLE IF EXISTS `user_roles`;
```
add newline

petkocfc commented on travel/src/main/resources/db/migration/V5__add_columns_user.sql:
```diff
@@ -0,0 +1,8 @@
+    ADD COLUMN IF NOT EXISTS `id_no` varchar(20) DEFAULT NULL,
+    ADD COLUMN IF NOT EXISTS `cost_center_id` bigint(20) NOT NULL,
+    ADD COLUMN IF NOT EXISTS `role_id` int(11) NOT NULL,
+    ADD CONSTRAINT `FK_roles_role_id_constraint` FOREIGN KEY (`role_id`) REFERENCES `roles` (`id`);
```
add newline

petkocfc commented on travel/src/main/java/com/tx/travel/service/AuthService.java:
```diff
@@ -25,55 +24,47 @@ public AuthService(final UserRepository userRepository, final RoleRepository rol
     final Optional<User> userByEmail = userRepository.findByEmail(email);
     if (userByEmail.isPresent()) {
-      throw new UsernameAlreadyExistsException(email);
+      throw new EmailAlreadyExistsException(email);
```
method is called `findByUsernameOrEmail` and we are returning exception `EmailAlreadyExistsException` should not work like this?

jovanlekic committed:
Updated methods names, wrote display names, deleted copy-pasted @columns from SignupRequest, fixed all suggestions from PR.

jovanlekic committed:
Merge remote-tracking branch 'origin/feature/jwt_and_user_model' into feature/jwt_and_user_model

# Conflicts:
#	travel/src/main/java/com/tx/travel/controller/AuthController.java
#	travel/src/main/java/com/tx/travel/payload/request/SignupRequest.java
#	travel/src/main/java/com/tx/travel/service/AuthService.java
#	travel/src/main/resources/db/migration/V4__drop_user_roles.sql
#	travel/src/main/resources/db/migration/V5__add_columns_user.sql
#	travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java

jovanlekic committed:
Created new exception, fixed reverted code in AuthController

jovanlekic committed:
Rebased into master, fixed test class AuthServiceImplementationTest

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -1,149 +1,131 @@
+        () -> sut.findByUsernameOrEmail("", email));
+  }
+
+  @DisplayName("given string admin" + " when role was searched" + " then return admin role")
```
split in separate rows

megicivovic commented on travel/src/test/java/com/tx/travel/service/AuthServiceImplementationTest.java:
```diff
@@ -1,149 +1,131 @@
+    assertEquals(result, role);
+  }
+
+  @DisplayName("given string mod" + " when role was searched" + " then return office manager role")
```
split in separate rows

megicivovic reviewed (commented):
Almost there, just be mindfull of the formatting!

jovanlekic committed:
Fixed indents, added dependencies

megicivovic reviewed (approved):
Amazing work! 🚀
